packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/__tests__/quanticColoredResultBadge.test.js

```diff
@@ -0,0 +1,180 @@
+import {createElement} from 'lwc';
+import QuanticColoredResultBadge from '../quanticColoredResultBadge';
+
+const defaultPrimaryColor = '#FFF7BA';
+const defaultSecondaryColor = '#E2B104';
+const defaultLabel = 'Case';
+
+const defaultResult = {
+  raw: {
+    fieldOne: 'valueOne',
+  },
+};
+
+const colorsToTest = [
+  {
+    primaryColor: '#DABFE9',
+    expectedSecondaryColor: 'hsl(279, 48.8%, 35.1%)',
+  },
+  {
+    primaryColor: '#EF233C',
+    expectedSecondaryColor: 'hsl(353, 86.4%, 5.7%)',
+  },
+  {
+    primaryColor: '03A06E',
+    expectedSecondaryColor: 'hsl(161, 96.3%, 80.0%)',
+  },
+  {
+    primaryColor: 'FFF',
+    expectedSecondaryColor: 'hsl(0, 0%, 52.0%)',
+  },
+];
+
+const invalidColors = [
+  {
+    primaryColor: '#ZZFFFF',
+    portion: 'red',
+  },
+  {
+    primaryColor: '#FFZZFF',
+    portion: 'green',
+  },
+  {
+    primaryColor: '#FFFFZZ',
+    portion: 'blue',
+  },
+];
+
+const invalidPropertiesError =
+  '"QuanticColoredResultBadge" requires either specified value for label or a result object with a fieldname to display correctly.';
```

Comment:
So you decided to put the logic into the badge component directly. What made you change your mind?

Reply:
Mainly I wanted to follow the same pattern used in the QuanticLabel component which has this `label vs field` logic, but on a second thought now I think you are right, having a simple component that will simply display the label passed to him is far better. And the logic to choose between displaying a label or field value will only be at a higher level, for example in the insight Panel it will be handled in the `ConfigurableResult` component.
Thank you for your suggestion I'll update this.👌

Reply:
Component simplified 👌 the badge will now just display  the text we're giving it.